### ISSUE TITLE
Installing on Raspberry PI 3 failed

### ISSUE THREAD

**Issue by JochemLangerak** [kservices/gBridge-docker#2]
Hi, 

After creating a docker-compose file using the generator (https://about.gbridge.io/dockergen/), i tried to use docker-compose up on my Raspberry pi 3 to start gBridge. I ended up with these kind of errors:

redis-worker_1  | standard_init_linux.go:207: exec user process caused "exec format error"
web_1           | standard_init_linux.go:207: exec user process caused "exec format error"

Than I tried to create the docker images myself on the pi cloning this repo and executed the following commands in the main folder:

docker build --tag=jl/gbridge-web --file ./docker-web/Dockerfile .
docker build --tag=jl/gbridge-web-fpm --file ./docker-web-fpm/Dockerfile .
docker build --tag=jl/gbridge-web-nginx --file ./docker-web-nginx/Dockerfile .
docker build --tag=jl/gbridge-redis-worker --file ./docker-redis-worker/Dockerfile .

I saw in this repo was a newer docker-compose.yml file which i used, changed the image names to my own, used hypriot/rpi-mysql:latest for mysql database and  arm32v7/redis:4 for cache (redis).

Using docker-compose up, i can now start the gBridge environment, but I have issues with the first command following the guide:

docker-compose exec web php artisan migrate

first i got an PATH error unknown. sh into the web container learned that php isn't recognized. I than installed php7.0-cli inside the web container, but I now got this error:

In Translator.php line 89:
  Parse error: syntax error, unexpected '?', expecting variable (T_VARIABLE)

Is there a possibility to get this working on a raspberry pi 3 or am i doing something wrong?

Best regards,

Jochem

**Comment by JochemLangerak** [comment]
Update: 
I went back to the image without all the extra php-cli installs and thought that maybe the command from the self hosted guide(https://doc.gbridge.io/selfHosted/hostItYourself.html#prepare-the-database):

docker-compose exec web php artisan migrate

should be changed in:

sudo docker-compose exec web-fpm php artisan migrate

Now, I do get a prompt "Enter yout E-Mail address". After that I got all sorts of SQL errors. Any suggestions would be welcome (maybe a different mysql docker image?). Here are the errors:

In Connection.php line 664:
  SQLSTATE[42000]: Syntax error or access violation: 1064 You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use nea
  r 'json null' at line 1 (SQL: alter table `trait` add `config` json null)

In PDOConnection.php line 63:
  SQLSTATE[42000]: Syntax error or access violation: 1064 You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use nea
  r 'json null' at line 1

In PDOConnection.php line 61:
  SQLSTATE[42000]: Syntax error or access violation: 1064 You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use nea
  r 'json null' at line 1

Best regards,

Jochem
